doc/Language/iterating.pod6

```diff
@@ -4,136 +4,321 @@
+my \dna = DNA.new('ACGTACGTT');
+
+say dna.map(*.join).join('|'); # OUTPUT: «GCA|TAC|TTG␤»
+say dna.map(*.join).join('|'); # OUTPUT: «␤»
 =end code
 
-We declare a C<DNA> class which does the two roles, C<Iterator> and C<Iterable>;
-the class will include a string that will be constrained to have a length that
-is a multiple of 3 and composed only of ACGT.
-
-Let us look at the
-C<pull-one> method. This one is going to be called every time a new iteration
-occurs, so it must keep the state of the last one. An C<$.index> attribute will
-hold that state across invocations; C<pull-one> will check if the end of the
-chain has been reached and will return the C<IterationEnd> constant provided by
-the role. Implementing this low-level interface, in fact, simplifies the
-implementation of the C<Iterable> interface. Now the iterator will be the object
-itself, since we can call C<pull-one> on it to access every member in turn;
-C<.iterator> will thus return just C<self>; this is possible since the object
-will be, at the same time, C<Iterable> and C<Iterator>.
-
-This need not always be the case, and in most cases C<.iterator> will have to
+Keeping the aforementioned caveats in mind, implementing this low-level
+interface simplifies the implementation of the C<Iterable> interface. Now the
+iterator will be the object itself, since we can invoke C<pull-one> on it to
+access every member in turn; C<iterator> will thus return just C<self>, which is
+possible since the object will be both an C<Iterable> and an C<Iterator>.
+
+This need not always be the case, and in most cases C<iterator> will have to
 build an iterator type to be returned (that will, for instance, keep track of
 the iteration state, which we are doing now in the main class), such as we did
-in the previous example; however, this example shows the minimal code needed to
-build a class that fulfills the iterator and iterable roles.
+in the previous section; however, this example shows the minimal code needed to
+build a class that fulfills both the iterator and iterable roles.
+
+=head1 Manual iteration
+
+Iterating constructs, such as C<for> loops, use iterators to iterate through
+collections. However, the programmer can also request values from an iterator
+one at a time without the aid of any iterating construct.
+
+You can get an iterator from any iterable, and you can use an iterator to manually
+loop over the iterable it came from. Thus, the way to manually iterate through an
+iterable is to get a hold of an iterator object by invoking the C<iterator> method
+and call the C<pull-one> method on said iterator to request a new item until
+C<IterationEnd> is returned. 
+
+=begin code
+my @languages = <perl python haskell ruby>;
+my $iter = @languages.iterator;
+
+say $iter.pull-one; #=> «perl␤» 
+say $iter.pull-one; #=> «python␤» 
+say $iter.pull-one; #=> «haskell␤» 
+say $iter.pull-one; #=> «ruby␤» 
+say $iter.pull-one; #=> «IterationEnd␤» 
+=end code
+
```

Comment:
In the manuals we should always reflect intent (#1748 ). This can be easily figured out from the syntax, and it's not an actual use case for pull-one (which will probably be used only internally). Not clear it fits within the title of the page, also.